.github/scripts/corresponding_source_report.py

```
@@ -0,0 +1,217 @@
+            break
+        url = expanded
+    return url
+
+
+def head_mirror_object(path: str) -> bool | None:
+    """True/False = object present/absent on the public mirror; None = probe
+    failed (network/5xx) — reported as unknown rather than missing."""
+    if not _SAFE_PATH.match(path):
+        return None
+    try:
+        p = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)